odp_sdk/raw_storage_client.py

```diff
@@ -0,0 +1,264 @@
+        file_meta: Optional[dict[str, any] | FileMetadataDto] = None,
+        filename: str = None,
+        mime_type: str = None,
+        contents: Optional[bytes | BytesIO] = None,
+        overwrite: bool = False,
+    ) -> FileMetadataDto:
+        """
+        Create a new file.
+
+        Args:
+            dataset_reference: Dataset manifest or name of dataset or UUID of dataset
+            file_meta: File metadata
+            filename: Optional way to specify the filename
+            mime_type: Optional way to specify the MIME type,
+            contents: File contents
+            overwrite: Overwrite if the file already exists
+
+        Returns:
+            The metadata of the newly created file
+        """
+        if isinstance(file_meta, FileMetadataDto):
+            file_meta_dto = file_meta
+        elif isinstance(file_meta, dict):
+            file_meta_dto = FileMetadataDto(**file_meta)
+        elif filename and mime_type:
+            file_meta_dto = FileMetadataDto(filename=filename, mime_type=mime_type)
+        else:
+            raise ValueError("You must provide either 'file_meta' or both 'filename' and 'mime_type'")
+
+        url = self._resolve_dataset_url(dataset_reference, endpoint=f"/{file_meta_dto.name}")
+        response = self.http_client.post(url, content=file_meta_dto.model_dump_json(exclude_unset=True))
+
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            if response.status_code == 401:
+                raise OdpValidationError("API argument error") from e
+            raise  # Unhandled error
+
+        file_meta = FileMetadataDto(**response.json())
+
+        if contents:
+            return self.update_file(file_meta, contents, overwrite)
+
+        return self.get_file_metadata(dataset_reference, file_meta_dto.name)
+
+    def download_file(
+        self, dataset_reference: str | ResourceDto | UUID, file: FileMetadataDto | str, save_path: str = None
+    ):
+        """
+        Download a file.
+
+        Args:
+            dataset_reference: Dataset manifest or name of dataset or UUID of dataset
+            file: File metadata or file name
+            save_path: File path to save the downloaded file to
+        """
+
+        filename = file.name
+
+        if isinstance(file, str):
+            filename = file
+
+        url = self._resolve_dataset_url(dataset_reference, endpoint=f"/{filename}")
+
+        response = self.http_client.get(url)
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            if response.status_code == 404:
+                raise OdpFileNotFoundError(f"File not found: {filename}") from e
+
+        if save_path:
+            with open(save_path, "wb") as file:
+                file.write(response.content)
+        else:
+            return response.content
+
+    def delete_file(self, dataset_reference: str | ResourceDto | UUID, filename: str) -> bool:
```

Comment:
Does this function have to return anything?

Reply:
It's returning the content if you dont want the sdk to store it to a file.

Reply:
The type-hint indicate a `bool`

Reply:
Ah you mean delete_file. I guess it should return a true/false if deletion went well? Or maybe raising exceptions will do this.

Reply:
Removed bool return value